### ISSUE TITLE
ReactorInstance reports an error that width cannot be inferred for valid C++ program

### ISSUE THREAD

**Issue by cmnrd** [lf-lang/lingua-franca#465]
This minimal program
```
target Cpp

reactor Foo {}

main reactor {
    foo_bank = new[{=42=}] Foo();
}
``` 
does not compile with lfc due to the following error being reported by ReactorInstance:
```
5 | main reactor {
6 |     foo_bank = new[{=42=}] Foo();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Cannot infer width.
  |
7 | }
``` 
However, since the C++ target supports widths to be given by target code, the program is actually correct and compiles fine when commenting out the corresponding check in ReactorInstance.xtend. This indicates that we need a mechanism to disable this check (and potentially others) for the C++ target, but I don't really know how to do this in a good way. Currently, the code in ReactorInstance is target agnostic, and I think it would be a good idea to keep it this way. Maybe we could introduce a parameter that indicates whether a width that cannot be inferred is problematic or not, but then we need to pass this parameter all the way down from the validator. I am also wondering whether we need to build the reactor instances at all for the C++ target, since it's not really used anywhere.

**Comment by edwardalee** [comment]
Would it work to just change this to a warning and move on?

**Comment by cmnrd** [comment]
That would be a quick fix, which I actually already committed in 422790a5c6a087827eb20e0fe159d9965e82a44e on the cpp-benchmarks branch (#406). This allows us to compile the benchmarks that rely on this feature. However, I don't think making the message a warning is a viable solution, because
1. In the C target, this condition should be an error, not a warning.
2. In the C++ target, the code is perfectly fine because the precise width is determined at runtime. Thus, a warning message in the compiler would be misleading.

**Comment by cmnrd** [comment]
@edwardalee Do you think it would be viable to just remove this check altogether? It seems that in the C target, the validator already checks whether the width can be determined and hence the check here should always succeed, but I might be missing something.

**Comment by edwardalee** [comment]
Yes, it may be OK to remove the check. However, there are potentially aspects of the resulting instance graph structure that will be incorrect. Consider a situation where a dependence between reactions exists or does not exist depending on what the width of a port is.  The C++ target does this dependence analysis in the runtime, so I guess it is OK in that case. Is there anything else where there is a reliance on such dependency analysis?  If not, I suggest just removing the check. But I do fear this will come back and bite us in the future.

**Comment by edwardalee** [comment]
This appears to be fixed in the scalability-banks branch. We can close this when that branch is merged.

**Comment by cmnrd** [comment]
This was fixed by #875.
